### ISSUE TITLE
Setting a mark while the cursor is on marked text could remove that mark

### ISSUE THREAD

**Issue by kristiandupont** [ryu1kn/vscode-text-marker#6]
This is the way I am used to using markers in most other editors: if the cursor is on text that is already marked, marking it again will remove the mark instead.

**Comment by ryu1kn** [comment]
If you don't select any text, TextMarker selects and marks/unmarks the word under the cursor. So if the marked text doesn't match the word under the cursor, TextMarker recognises the word as a new pattern and marks it instead of unmarking the marked text.

```
[marked text]       # "marked text" is marked
↓
[marked tex|t]      # Place the cursor on "text", and execute "mark/unmark text" command
↓
[marked [text]]     # "text" is recognised as a new pattern
                    # and both "text" and "marked text" will be marked
```

If you're seeing this behaviour, this is actually what I'm expecting.

**Comment by kristiandupont** [comment]
Yeah, I've noticed that. It's not my personal preference, but this *is* your project so obviously it's up to you :-) Thank you for the extension by the way, it seems to work nicely!

**Comment by ryu1kn** [comment]
I actually haven't thought about that way of unmarking text, but I can see the advantages of it.

As I'm not too sure if I like it more, I'm not gonna change it right now. But thank you for your suggestion ;)

**Comment by ryu1kn** [comment]
After introduced highlighting with regex, I like the idea you suggested better.
With this way, we can remove a mark that is specified by regex.

**Comment by ryu1kn** [comment]
Published as v0.9.0! Thank you for your suggestion @kristiandupont 👍

**Comment by kristiandupont** [comment]
Awesome, looking forward to using it! :-)
